src/pke/lib/scheme/bgvrns/bgvrns-parametergeneration.cpp

```diff
@@ -213,11 +214,13 @@ std::pair<std::vector<NativeInteger>, uint32_t> ParameterGenerationBGVRNS::compu
                 "Change parameters! Try reducing the number of additions per level, "
                 "number of key switches per level, or the digit size. We cannot support moduli greater than 60 bits.");
         }
-        totalModSize += extraModSize;
-        moduliQ[numPrimes] = FirstPrime<NativeInteger>(extraModSize, cyclOrder);
+
+        moduliQ[numPrimes] = FirstPrime<NativeInteger>(extraModSize - 1, cyclOrder);
```

Comment:
Why not LastPrime?

Reply:
a few unit tests would request LastPrime(12, 512), LastPrime(10, 512), LastPrime(10, 1024), LastPrime(10, 16384). These all seem unrealistic. We can change the unit tests parameters.

Reply:
these are not realistic. Increase the bit sizes in the tests to first ones that pass.

Reply:
keeping FirstPrime<NativeInteger>(extraModSize - 1, cyclOrder);